source/lex.tex

```diff
@@ -1250,135 +1268,179 @@
-begins with \tcode{u8}, such as \tcode{u8'w'},
-\indextext{prefix!\idxcode{u8}}%
-is a \grammarterm{character-literal} of type \tcode{char8_t},
-known as a \defn{UTF-8 character literal}.
-The value of a UTF-8 character literal
-is equal to its ISO/IEC 10646 code point value,
-provided that the code point value
-can be encoded as a single UTF-8 code unit.
-\begin{note}
-That is, provided the code point value is in the range $[0, \mathrm{7F}]$ (hexadecimal).
-\end{note}
-If the value is not representable with a single UTF-8 code unit,
-the program is ill-formed.
-A UTF-8 character literal containing multiple \grammarterm{c-char}{s} is ill-formed.
-
-\pnum
-\indextext{literal!character!UTF-16}%
+\indextext{type!\idxcode{char8_t}}%
 \indextext{type!\idxcode{char16_t}}%
-A \grammarterm{character-literal} that
-begins with the letter \tcode{u}, such as \tcode{u'x'},
-\indextext{prefix!\idxcode{u}}%
-is a \grammarterm{character-literal} of type \tcode{char16_t},
-known as a \defn{UTF-16 character literal}.
-The value of a UTF-16 character literal
-is equal to its ISO/IEC 10646 code point value,
-provided that the code point value is
-representable with a single 16-bit code unit.
+\indextext{type!\idxcode{char32_t}}%
+\indextext{wide-character}%
+\indextext{type!\idxcode{wchar_t}}%
+A \defnx{non-encodable character literal}{literal!character!non-encodable}
+is a \grammarterm{character-literal}
+whose \grammarterm{c-char-sequence} consists of a single \grammarterm{c-char}
+that is not a \grammarterm{numeric-escape-sequence} and
+that specifies a character
+that either lacks representation in the literal's associated character encoding
+or that cannot be encoded as a single code unit.
+A \defnadj{multicharacter}{literal} is a \grammarterm{character-literal}
+whose \grammarterm{c-char-sequence} consists of
+more than one \grammarterm{c-char}.
+The \grammarterm{encoding-prefix} of
+a non-encodable character literal or a multicharacter literal
+shall be absent or \tcode{L}.
+Such \grammarterm{character-literal}s are conditionally-supported.
+
+\pnum
+The kind of a \grammarterm{character-literal},
+its type, and its associated character encoding
+are determined by
+its \grammarterm{encoding-prefix} and its \grammarterm{c-char-sequence}
+as defined by \tref{lex.ccon.literal}.
+The special cases for
+non-encodable character literals and multicharacter literals
+take precedence over their respective base kinds.
 \begin{note}
-That is, provided the code point value is in the range $[0, \mathrm{FFFF}]$ (hexadecimal).
+The associated character encoding for ordinary and wide character literals
+determines encodability,
+but does not determine the value of
+non-encodable ordinary character literals or ordinary multicharacter literals.
+The examples for non-encodable ordinary and wide character literals assume that
+the specified character lacks representation in the execution character set, or
+that encoding it would require more than one code unit.
```

Comment:
Taking the footnotes out of context doesn't work. The footnotes talk about "examples" - i only could figure out what examples the wording was talking about by going back to the original and seeing where the footnotes appeared inn the table.  Please leave as footnotes, and stick closer to the original wording.

Reply:
Real footnotes inside tables don't work LaTeX-wise.
I've added an explicit reference to the table; I think this improves the "where to look" situation.

Reply:
That helps a lot, thank you!

Reply:
Two more suggestions:

1. Shorten "does not determine the value of non-encodable ordinary character literals or ordinary multicharacter literals" to "does not determine the value of non-encodable ordinary or wide character literals".

2. Add "lacks representation in the execution <ins>or wide-execution</ins> character set<ins>, respectively</ins>, or". Otherwise we lose the distinction between footnotes Y2 and Y4.

If you agree, just say so, and I will make the edits as part of the merge (no need to push any commits).

Reply:
@burblebee, @jensmaurer ^^^.

Reply:
This is called the "execution wide-character set", so the phrasing becomes a bit more verbose.
Also, we can have ordinary or wide multicharacter literals. See the update ("fixup").

Reply:
Ah yes, excellent, thanks!